coordinator/core/manifest.go

```diff
@@ -201,6 +203,34 @@ func (m Manifest) Check(ctx context.Context, zaplogger *zap.Logger) error {
 	return nil
 }
 
+// CheckUpdate checks if the manifest is consistent and only contains supported values.
+func (m Manifest) CheckUpdate(ctx context.Context, originalPackages map[string]quote.PackageProperties, alreadyUpdatedPackages map[string]quote.PackageProperties) error {
+	if len(m.Packages) <= 0 {
+		return errors.New("no packages defined")
+	}
+
+	// Check if manifest update contains values which we normally should not update
+	for packageName, singlePackage := range m.Packages {
+		if singlePackage.Debug != false || singlePackage.UniqueID != "" || singlePackage.SignerID != "" || singlePackage.ProductID != nil {
+			return errors.New("update manifest contains unupdatable values")
+		}
+
+		// Checks based on the original manifest
+		if *singlePackage.SecurityVersion < *originalPackages[packageName].SecurityVersion {
```

Comment:
singlePackage.SecurityVersion might be nil
originalPackages[packageName] might not exist
originalPackages[packageName].SecurityVersion might be nil